docs/pages/tutorials/cpp.rst

```diff
@@ -34,61 +38,97 @@ We **strongly recommend** to also call the ``set_wavemap_target_properties`` fun
 
 Code examples
 *************
-
 In the following sections, you'll find sample code for common tasks. If you'd like to request examples for additional tasks or contribute new examples, please don't hesitate to `contact us <https://github.com/ethz-asl/wavemap/issues>`_.
 
-Serialization
-=============
+Serializing maps
+================
+In this section, we'll demonstrate how to serialize and deserialize maps using wavemap's lightweight and efficient binary format. This format is consistent across wavemap's C++, Python, and ROS interfaces. For instance, you can create maps on a robot with ROS and later load them into a rendering engine plugin that only depends on wavemap's C++ library.
 
-Files
------
-Saving maps to files:
+Binary files
+------------
+Maps can be saved to disk with
 
 .. literalinclude:: ../../../examples/cpp/io/save_map_to_file.cc
     :language: c++
 
-Loading maps from files:
+.. _cpp-code-examples-read-map:
+
+and read using
 
 .. literalinclude:: ../../../examples/cpp/io/load_map_from_file.cc
     :language: c++
 
+Byte streams
+------------
+We also provide an alternative, lower-level interface to convert maps to (byte) streams
+
+.. literalinclude:: ../../../examples/cpp/io/save_map_to_stream.cc
+    :language: c++
+
+and read them with
+
+.. literalinclude:: ../../../examples/cpp/io/load_map_from_stream.cc
+    :language: c++
+
+
 Queries
 =======
+In this section, we illustrate how you can query the map and classify whether a point or region of interest is occupied.
+
+Node indices
+------------
+The map models the environment by filling it with cubes of variable sizes, arranged as the nodes of an octree. Node indices are defined as integer [X, Y, Z, height] coordinates, whose XYZ values correspond to the node's position in the octree's grid at the given *height*, or level in the tree. Height 0 corresponds to the map's maximum resolution, and the grid resolution is halved for each subsequent height level.
```

Comment:
Did you want to include an image of the concept here to make it easy to understand for people?